pkg/manager/internal.go

```diff
@@ -544,6 +549,10 @@ func (cm *controllerManager) engageStopProcedure(stopComplete <-chan struct{}) e
 		cm.runnables.LeaderElection.startOnce.Do(func() {})
 		cm.runnables.LeaderElection.StopAndWait(cm.shutdownCtx)
 
+		// Stop the warmup runnables
+		cm.logger.Info("Stopping and waiting for warmup runnables")
+		cm.runnables.Warmup.StopAndWait(cm.shutdownCtx)
```

Comment:
Just to confirm my understanding. This will:
* cancel the context that was used to call Warmup()
* wait until all Warmup funcs returned

On the happy path where all Warmup funcs completed during controller startup this will do effectively nothing

Reply:
Yes. In the happy path `StopAndWait` would just block until the waitgroup tracking runnables has counter == 0 which is will because the Warmup funcs have already completed.

In the unhappy path `StopAndWait` would cancel the context and since the controller's implementation of warmup respects the calling context, it early exits without waiting for warmup to finish which functionally makes sense to be despite it only `Stopping` and not really `Waiting`